pyiceberg/table/upsert_util.py

```diff
@@ -36,7 +37,17 @@ def create_match_filter(df: pyarrow_table, join_cols: list[str]) -> BooleanExpre
     if len(join_cols) == 1:
         return In(join_cols[0], unique_keys[0].to_pylist())
     else:
-        return Or(*[And(*[EqualTo(col, row[col]) for col in join_cols]) for row in unique_keys.to_pylist()])
+        filters: List[BooleanExpression] = [
+            cast(BooleanExpression, And(*[EqualTo(col, row[col]) for col in join_cols])) for row in unique_keys.to_pylist()
+        ]
+
+        if not filters:
+            return In(join_cols[0], [])
+
+        if len(filters) == 1:
+            return filters[0]
+
+        return functools.reduce(lambda a, b: Or(a, b), filters)
```

Comment:
What do you think of the following:
```suggestion
        if len(filters) == 0:
            return AlwaysTrue()
        elif len(filters) == 1:
            return filters[0]
        else:
            return functools.reduce(lambda a, b: Or(a, b), filters)
```

Reply:
I think , Returning In(join_cols[0], []) is correct because it results in a predicate that is always false, so no target row matches the update condition.
Returning AlwaysTrue() would be wrong because it would match every row in the target, causing an unintended update of all rows.
What do you thing?

Reply:
You are 💯 right, this was before my morning coffee:

```
➜  iceberg-python git:(main) ✗ python3
Python 3.10.14 (main, Mar 19 2024, 21:46:16) [Clang 15.0.0 (clang-1500.3.9.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from pyiceberg.expressions import In
>>> In('vo', [])
AlwaysFalse()
```

Reply:
@omkenge I've updated the suggestion, I think `AlwaysFalse()` is easier to interpret, what do you think?